### DISCUSSION TITLE
What query parameters does Remix send?

### DISCUSSION THREAD

**Discussion by liegeandlief** [remix-run/remix#2146] [Q&A] [answered]
I want to implement caching in CloudFront based upon only those query parameters I know will alter the response. This is so that things like Google Analytics query parameters don't result in cache misses, and also to prevent malicious bypassing of the cache by including random query parameters.

I know the names of query parameters which my application makes use of directly but need to know what query parameters Remix might send so that I can include them in the cache key.

I have noticed that when calling loaders and actions a `_data` parameter is sent. Are there any others? Also I think it would be helpful if this were documented somewhere as I think caching based on known query parameters is a fairly common scenario.

Thanks!

**Comment by sergiodxa** [comment, marked as answer]
Only `?_date` for data requests, the rest are your own or some library you may use.

**Comment by FStell01** [comment]
Stop messaging me
